org.eclipse.jdt.apt.pluggable.tests/src/org/eclipse/jdt/apt/pluggable/tests/processors/buildertester/Issue565Processor.java

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Kamil Krzywanski
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Kamil Krzywanski - initial creation if Interesection type and Implementation
+ *******************************************************************************/
+package org.eclipse.jdt.apt.pluggable.tests.processors.buildertester;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.*;
+import javax.lang.model.util.SimpleTypeVisitor8;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+@SupportedAnnotationTypes("targets.issue565.Annotation565")
+@SupportedSourceVersion(SourceVersion.RELEASE_6)
+public class Issue565Processor extends AbstractProcessor {
+	private static boolean status = false;
+
+	@Override
+	public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+
+		String classpath = System.getProperty("java.class.path");
+		String[] classPathValues = classpath.split(File.pathSeparator);
+		for (String classPath: classPathValues) {
+			System.out.println(classPath);
+		}
+
+		for (Element element : roundEnv.getElementsAnnotatedWith(annotations.stream().findAny().get())) {
+				if (element instanceof TypeElement) {
```

Comment:
I am wondering we make the testing parameters a bit more elaborate? Looks like we aren't doing much with the visiting elements and all we check for is that getElementsAnnotatedWith() doesn't crash. We don't seem to be using what the keyBuilder is doing. Also, this for loop is not indented properly.